docs/references/architecture/adr-117-multi-curve-support.md

```diff
@@ -0,0 +1,245 @@
+# ADR 117: Cryptography: Multi curve support
+
+## Change log
+
+* July 1st 2024: Initial proposal (Zondax AG: @raynaudoe @juliantoledano @jleni @educlerici-zondax @lucaslopezf)
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes the implementation of [ADR-001-crypto-provider](https://github.com/cosmos/crypto/blob/main/docs/architecture/adr-001-crypto-provider.md). By refactoring the current `PrivValidator` implementations and replacing them with `CryptoProvider`s, we will allow for "pluggable cryptography" that enables a more modular design and future-proofs crypto tools.
+
+We will be extensively referencing concepts like CryptoProviders in this ADR. Please check the full ADR describing them [here](https://github.com/cosmos/crypto/blob/main/docs/architecture/adr-001-crypto-provider.md#crypto-provider).
+
+## Context
+
+The introduction of multi-curve support, by implementing the `CryptoProvider` interface and its components, in the CometBFT cryptographic package offers significant advantages. By not being restricted to a single cryptographic curve, developers can choose the most appropriate curve based on security, performance, and compatibility requirements. This flexibility enhances the application's ability to adapt to evolving security standards and optimizes performance for specific use cases, helping to future-proof the codebase's cryptographic capabilities.
+
+## Decision
+
+We will:
+
+* Define types and interfaces as the code attached
+* Refactor existing code into new structure and interfaces (more details in the next section)
+* Implement Unit Tests to ensure no backward compatibility issues
+* Establish a migration path where the previous implementation will coexist with the new one, ensuring no sudden breaking changes occur
+
+
+## Detailed Design
+
+The implementation of ADR-001 in CometBFT will require a medium-level refactor of the codebase. Below we describe the proposed strategy to perform such implementation, which is:
+
+* Add a new `PrivValidator` implementation that uses the `CryptoProvider` interface underneath.
+
+* New directories reorganization:
+Implementations of crypto providers (previously `Privval` implementations) should be in their own directory. See the [Directories reorganization](#directories-reorganization) section for more details.
+
+* **[Optional]** Adding [Keyring](https://github.com/cosmos/cosmos-sdk/blob/main/crypto/keyring/keyring.go) and [Record](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/crypto/keyring/v1/record.proto) for storing and loading providers.
+
+* **[Optional]** Use `Keyring` to load and instantiate validators when booting up a node.
```

Comment:
Why would we want to load validators from the keyring? For additional safety?

Reply:
In short, yes. Is an optional feature to improve security